### ISSUE TITLE
GitHub API limitation issue

### ISSUE THREAD

**Issue by Kyle-Ye** [swift-server/swiftly#61]
I tried to migrate from swiftbox to swiftly.

1. Install it and config it.
```shell
curl -L https://swift-server.github.io/swiftly/swiftly-install.sh | bash
> 1
. $HOME/.local/share/swiftly/env.sh
```

2. Try to install swift toolchain and I'll always get a 403 Forbidden result.
```shell
➜  kyle swiftly        
OVERVIEW: A utility for installing and managing Swift toolchains.

USAGE: swiftly <subcommand>

OPTIONS:
  --version               Show the version.
  -h, --help              Show help information.

SUBCOMMANDS:
  install                 Install a new toolchain.
  use                     Set the active toolchain.
  uninstall               Remove an installed toolchain.
  list                    List installed toolchains.

  See 'swiftly help <subcommand>' for detailed help.
➜  kyle swiftly list
Installed release toolchains
----------------------------

Installed snapshot toolchains
-----------------------------
➜  kyle swiftly install
Error: Missing expected argument '<version>'
Help:  <version>  The version of the toolchain to install.
Usage: swiftly install <version> [--token <token>]
  See 'swiftly install --help' for more information.
➜  kyle swiftly install latest
Fetching the latest stable Swift release...
Error: received status "403 Forbidden" when reaching https://api.github.com/repos/apple/swift/releases?per_page=100&page=1: {"message":"API rate limit exceeded for xxx. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)","documentation_url":"https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting"}

➜  kyle swiftly install 5.9   
Fetching the latest stable Swift 5.9 release...
Error: received status "403 Forbidden" when reaching https://api.github.com/repos/apple/swift/releases?per_page=100&page=1: {"message":"API rate limit exceeded for xxx. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)","documentation_url":"https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting"}
```

**Comment by Kyle-Ye** [comment]
The issue is due to the IP I use has reached GitHub limitation. After I changed to another IP to access GitHub API the issue is gone.

**Comment by patrickfreed** [comment]
Yeah this is a known limitation of using the GitHub API. To work around it, you can generate a no-privilege token and then pass it into swiftly via the `--token` argument or via the `SWIFTLY_GITHUB_TOKEN` environment variable. There's a [note](https://github.com/swift-server/swiftly#specifying-a-github-access-token) about this in the README, though I admit it is not the easiest to find.

**Comment by kkebo** [comment]
Does `swiftly update` have `--token`?

**Comment by kkebo** [comment]
> Does swiftly update have --token?

No, but it may be fixed by #172.
